packages/vats/src/core/lib-boot.js

```diff
@@ -202,31 +200,12 @@ export const makeBootstrap = (
     },
 
     //#region testing supports
-    messageVat: async ({ name, methodName, args = [] }) => {
-      const vat = vatData.get(name) || Fail`unknown vat name: ${q(name)}`;
-      const { root } = vat;
-      const decodedArgs = args.map(decodePassable);
-      const result = await E(root)[methodName](...decodedArgs);
-      return encodePassable(result);
-    },
-    messageVatObject: async ({ presence, methodName, args = [] }) => {
-      const object = decodePassable(presence);
-      const decodedArgs = args.map(decodePassable);
-      const result = await E(object)[methodName](...decodedArgs);
-      return encodePassable(result);
-    },
-    /* Like `messageVatObject` but does not await return value */
-    messageVatObjectSendOnly: ({ presence, methodName, args = [] }) => {
-      const object = decodePassable(presence);
-      const decodedArgs = args.map(decodePassable);
-      void E(object)[methodName](...decodedArgs);
-    },
-    awaitVatObject: async ({ presence, path = [], rawOutput = false }) => {
-      let value = await decodePassable(presence);
+    awaitVatObject: async (presence, path = []) => {
```

Comment:
Let's see, this removes three utility methods (`messageVat`/`messageVatObject`/`messageVatObjectSendOnly`) from the vat-bootstrap root object, and changes the behavior of `awaitVatObject` to accept (promise for) some record, instead of the `encodePassable` encoding of the same (and does not `encodePassable` the result).

The mainnet chain vat-bootstrap will still have them, and it's `awaitVatObject` will continue to use `decodePassable`/`encodePassable`. I want to think about what that difference means for both tests and mainnet CORE_EVAL proposals that need to do some sort of emergency access.

@dckc What did you have in mind for emergency access (to e.g. the produce/consume PromiseSpace)? It looks like we could have cosmic-swingset code do `controller.queueToVatRoot('bootstrap', 'messageVat', ..)` to make it talk to some other vat (e.g. zoe) and get the results stashed in the `makeEncodePassable()` table, then a `controller.queueToVatRoot('bootstrap', `messageVatObject`, ..)` could target that result and send it a message.

Without them, the cosmic-swingset code would need to figure out the exact `kref` of interest and use the new `controller.queueToVatObject()` to target it.

It's a moot question for mainnet, of course, since the bootstrap vat isn't being upgraded. But developers of emergency fixes will need to take a very different approach depending upon whether they're targeting mainnet (almost certainly) or targeting a new chain whose bootstrap lacks those methods.

@dckc I imagine that emergency fixes are better implemented by a CORE_EVAL which evalutes a whole bundle inside the existing bootstrap vat, where it can make normal `E(produce.whatever).stuff()` calls, and actually `await` the results, and make multiple calls (we should have a name for this fancy interaction, I know let's call it a "program" :-). So using `controller.queueToVatwhatever` from the outside isn't the preferred emergency tool.

I think I've talked myself into being ok with it. And it certainly leaves the codebase looking simpler.

Reply:
> The mainnet chain vat-bootstrap will still have them, and it's awaitVatObject will continue to use decodePassable/encodePassable. I want to think about what that difference means for both tests and mainnet CORE_EVAL proposals that need to do some sort of emergency access.

That method is explicitly marked TEST ONLY and so it frightens me a lot to have it in production.  I don't know that it does anything you'd want to allow on mainnet; I really want to just remove it.  However, when I tried to remove it from `vats/src/core/lib-boot.js` (as I did with all the message relay methods) it turned out there was one test that used the (I think) main chain config instead of the test config that boots with `SwingSet/tools/bootstrap-relay.js`.  I suspect that one test could be fixed and then we could remove the method, but I didn't want to spend the time digging to puzzle it out.

Reply:
Also, the relay methods that addressed vats by name (very non-ocap, that) were weird.  Normal `queueToVatRoot` uses the kernel's vat name table, which is statically determined at boot time from the config definition.  The relay methods in `bootstrap-relay` (but **NOT** the ones in `lib-boot`, which also uses the same static lookup that `queueToVatRoot` uses) used a dynamic vat name table, but that table is only updated by vats created via the special test-mode `createVat` bootstrap vat method, not by dynamic vat creations in general, so it was really only suitable for testing anyway.

Reply:
> Let's see, this removes three utility methods (`messageVat`/`messageVatObject`/`messageVatObjectSendOnly`) from the vat-bootstrap root object ...
> @dckc What did you have in mind for emergency access (to e.g. the produce/consume PromiseSpace)?

Only `consumeItem`, `produceItem`, and `resetItem`, arising from #5740

The `//#region testing supports` comment is there to say that `messageVat` and friends were never intended for production use.

> @dckc I imagine that emergency fixes are better implemented by a CORE_EVAL...

Usually, yes. But #5740 imagines adding new devices, and I don't think we can do that with only a CORE_EVAL.

Reply:
Remembering that the code on chain is different seems tricky.

How about a comment with a BIG ASCII ART BANNER to the effect of:

**The agoric-3 mainnet does not run this version of the bootstrap vat.
A mainnet node will show v1:hash-bootstrap in `ps`, indicating
that code from RELEASE/HASH is running.**

Ideally we'd reify the "bootstrap is hard to upgrade" issue in github and cite it there too.